src/sshfs_mount/sshfs_server.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <QStringList>
+
+#include "../ssh/ssh_client_key_provider.h" // FIXME
+#include <multipass/exceptions/sshfs_missing_error.h>
+#include <multipass/logging/log.h>
+#include <multipass/logging/standard_logger.h>
+#include <multipass/ssh/ssh_session.h>
+#include <multipass/sshfs_mount/sshfs_mount.h>
+
+#include <signal.h>
+#include <sys/prctl.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+using namespace std;
+
+namespace
+{
+unordered_map<int, int> deserialise_id_map(const char* in)
+{
+    unordered_map<int, int> id_map;
+    QString input(in);
+    auto maps = input.split(',', QString::SkipEmptyParts);
+    for (auto map : maps)
+    {
+        auto ids = map.split(":");
+        if (ids.count() != 2)
+        {
+            cerr << "Incorrect ID mapping syntax";
+            continue;
+        }
+        bool ok1, ok2;
+        int from = ids.first().toInt(&ok1);
+        int to = ids.last().toInt(&ok2);
+        if (!ok1 || !ok2)
+        {
+            cerr << "Incorrect ID mapping ids found, ignored" << endl;
+            continue;
+        }
+        id_map.insert({from, to});
+    }
+    return id_map;
+}
+} // namespace
+
+int main(int argc, char* argv[])
+{
+    prctl(PR_SET_PDEATHSIG, SIGHUP); // ensure if parent dies, this process gets the SIGHUP signal
+
+    if (argc != 8)
```

Comment:
I'm wondering if we should make this a `QCoreApplication` and use proper arg handling???  I know this is a _private_ process only for `multipassd`'s consumption, but it seems a bit more cleaner to have some proper argument handling instead of relying on an arbitrary order of arguments passed in.  That said, I suppose they are positional arguments and really wouldn't be any different even if we made this a `QCoreApplication` with its positional arguments.

I'm kind on the fence with this one :grin:

Reply:
I don't see any real value in doing proper argument handling. As you say, it's not for user consumption. If you fall off the fence at some point, let me know which side you end up on :)

Reply:
I guess I'm falling off the fence on the side of leaving it as-is :grin: